djangocms_navigation/views.py

```diff
@@ -32,8 +32,18 @@ def get(self, request, *args, **kwargs):
         if not is_model_supported(self.menu_content_model, model):
             return HttpResponseBadRequest()
 
+        queryset_data = self.get_data()
+
+        # Removing unpublished pages from queryset
+        if model == Page:
+            queryset_data = [
+                page for page in queryset_data
+                if getattr(page.get_title_obj().versions.first(), "state", None)
```

Comment:
add `from djangocms_versioning.constants import ARCHIVED, UNPUBLISHED` at the top of the page.

update below codes to include checking  `archived`.
`
if not getattr(page.get_title_obj().versions.first(), "state", None)
in [ ARCHIVED, UNPUBLISHED ]
`

If you check the unpublished page '**Triage tool test**'  in `EU WPFH Multisite NP`, you can find the `get_title_obj` will return previously published pagecontent instance (see below screenshot), but now its state is `archived`. so I think that return of `archived` page is also meaningless. 

![image](https://github.com/FidelityInternational/djangocms-navigation/assets/100244/f3323e59-f48f-4850-8490-e4b1268b65da)

Reply:
@joshyu Can you please check now?